libs/x-content/src/main/java/org/elasticsearch/xcontent/DotExpandingXContentParser.java

```diff
@@ -71,6 +71,31 @@ protected XContentParser delegate() {
         }
     }
 
+    private static String[] splitAndValidatePath(String fullFieldPath) {
+        if (fullFieldPath.isEmpty()) {
+            throw new IllegalArgumentException("field name cannot be an empty string");
+        }
+        if (fullFieldPath.contains(".") == false) {
+            return new String[] { fullFieldPath };
+        }
+        String[] parts = fullFieldPath.split("\\.");
+        if (parts.length == 0) {
+            throw new IllegalArgumentException("field name cannot contain only dots");
+        }
+        for (String part : parts) {
+            // check if the field name contains only whitespace
+            if (part.isEmpty()) {
+                throw new IllegalArgumentException("object field cannot contain only whitespace: ['" + fullFieldPath + "']");
```

Comment:
@romseygeek is the mention of "object" in this error and the next one accurate? Don't we split any path regardless of the token we are reading from the parser?